packages/server/src/utils.ts

```diff
@@ -84,8 +93,12 @@ export function normalizeNodeRequest(
   nodeRequest: NodeRequest,
   RequestCtor: typeof Request,
 ): Request {
+  // TODO: this is a main issue!
+  // const rawRequest: NodeRequest | IncomingMessage | Http2ServerRequest
   const rawRequest = nodeRequest.raw || nodeRequest.req || nodeRequest;
```

Comment:
const rawRequest: NodeRequest | Http2ServerRequest | IncomingMessage this is an issue, here you had created a union `NodeRequest | Http2ServerRequest | IncomingMessage`, and rawRequest is passed down into functions.